File: support_tools/swigwin-3.0.12/Examples/test-suite/go/rename_simple_runme.go
package main

import "fmt"
import . "./rename_simple"

func main() {
	s := NewNewStruct()
	check(111, s.GetNewInstanceVariable(), "NewInstanceVariable")
	check(222, s.NewInstanceMethod(), "NewInstanceMethod")
	check(333, NewStructNewStaticMethod(), "NewStaticMethod")
	check(444, GetNewStructNewStaticVariable(), "NewStaticVariable")
	check(555, NewFunction(), "NewFunction")
	check(666, GetNewGlobalVariable(), "NewGlobalVariable")

	s.SetNewInstanceVariable(1111)
	SetNewStructNewStaticVariable(4444)
	SetNewGlobalVariable(6666)

	check(1111, s.GetNewInstanceVariable(), "NewInstanceVariable")
	check(4444, GetNewStructNewStaticVariable(), "NewStaticVariable")
	check(6666, GetNewGlobalVariable(), "NewGlobalVariable")
}

func check(expected, actual int, msg string) {
	if expected != actual {
		panic("Failed: Expected: " + fmt.Sprint(expected) +
			" actual: " + fmt.Sprint(actual) + " " + msg)
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/csharp/rename_simple_runme.cs

using System;
using rename_simpleNamespace;

public class rename_simple_runme {

  public static void Main() {
    NewStruct s = new NewStruct();
    check(111, s.NewInstanceVariable, "NewInstanceVariable");
    check(222, s.NewInstanceMethod(), "NewInstanceMethod");
    check(333, NewStruct.NewStaticMethod(), "NewStaticMethod");
    check(444, NewStruct.NewStaticVariable, "NewStaticVariable");
    check(555, rename_simple.NewFunction(), "NewFunction");
    check(666, rename_simple.NewGlobalVariable, "NewGlobalVariable");

    s.NewInstanceVariable = 1111;
    NewStruct.NewStaticVariable = 4444;
    rename_simple.NewGlobalVariable = 6666;

    check(1111, s.NewInstanceVariable, "NewInstanceVariable");
    check(4444, NewStruct.NewStaticVariable, "NewStaticVariable");
    check(6666, rename_simple.NewGlobalVariable, "NewGlobalVariable");
  }

  public static void check(int expected, int actual, string msg) {
    if (expected != actual)
      throw new Exception("Failed: Expected: " + expected + " actual: " + actual + " " + msg);
  }
}

